src/metabase/upload.clj

```diff
@@ -214,35 +226,33 @@
      :generated-columns (ordered-map/ordered-map :id ::auto-incrementing-int-pk)}))
 
 (defn- rows->schema
+  "rows should be a lazy-seq"
   [header rows]
   (let [normalized-header (->> header
                                (map normalize-column-name)
                                (mbql.u/uniquify-names)
                                (map keyword))
-        column-count      (count normalized-header)]
+        column-count      (count normalized-header)
+        settings          (upload-parsing/get-settings)]
```

Comment:
Now we get `settings` here, and pass it to `row->types`, which passes it to `value->type`